data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,150 @@
+
+- `autocommit_interval` (Optional) : The frequency in seconds that the consumer offsets are auto-committed to Kafka. Defaults to `1s`.
+
+- `session_timeout` (Optional) : The timeout used to detect client failures when using Kafka's group management. It is used for the rebalance.
+
+- `max_retry_attempts` (Optional) : The maximum attempts to retry a failed write operation to the BUFFER. Defaults to `INFINITE`.
+
+- `max_retry_delay` (Optional) : By default the Kafka source will retry for every 1 second when there is a buffer write error. Defaults to `1s`. 
+
+- `auto_offset_reset` (Optional) : automatically reset the offset to the earliest/latest offset. Defaults to `earliest`.
+
+- `thread_waiting_time` (Optional) : It is the time for thread to wait until other thread completes the task and signal it.
+
+- `max_record_fetch_time` (Optional) : maximum time to fetch the record from the topic.
+Defaults to `4s`.
+
+- `heart_beat_interval` (Optional) : The expected time between heartbeats to the consumer coordinator when using Kafka's group management facilities. Defaults to `1s`.
+
+- `buffer_default_timeout` (Optional) :  The maximum time to write data to the buffer. Defaults to `1s`.
+
+- `fetch_max_bytes` (Optional) : The maximum record batch size accepted by the broker. 
+Defaults to `52428800`.
+
+- `fetch_max_wait` (Optional) : The maximum amount of time the server will block before answering the fetch request if there isn't sufficient data to immediately satisfy the requirement. Defaults to `500`.
+
+- `fetch_min_bytes` (Optional) : The minimum amount of data the server should return for a fetch request. Defaults to `1`.
+
+- `retry_backoff` (Optional) : The amount of time to wait before attempting to retry a failed request to a given topic partition.  Defaults to `5s`.
+
+- `max_poll_interval` (Optional) : The maximum delay between invocations of poll() when using consumer group management. Defaults to `1s`.
+
+- `consumer_max_poll_records` (Optional) : The maximum number of records returned in a single call to poll(). Defaults to `1s`.
+
+### <a name="schema_configuration">Schema Configuration</a>
+
+- `key_deserializer` (Optional) : Deserialize a record value from a bytearray into a String. Defaults to `org.apache.kafka.common.serialization.StringDeserializer`.
+
+- `value_deserializer` (Optional) : Deserialize a record key from a bytearray into a String. Defaults to `org.apache.kafka.common.serialization.StringDeserializer`.
+
+- `schema_type` (Optional) : The type of schema format reading from the broker. This Kafka Consumer plugin supports String and json schema types. Defaults to `plaintext`.
+
+- `record_type` (Optional) : The type of record format reading from the broker. This Kafka Consumer plugin supports String and json record types. Defaults to `plaintext`.
+
+### <a name="ssl_configuration">SSL Auth Configuration</a>
```

Comment:
Are all these values base64 encoded? they should be to obscure secrets from plain sight.